### ISSUE TITLE
7.1.0: test failure in test_assertrewrite.py

### ISSUE THREAD

**Issue by bnavigator** [pytest-dev/pytest#9761]
- [x] a detailed description of the bug or problem you are having
- [x] ~~output of `pip list` from the virtual environment you are using~~ see rpm package versions
- [x] pytest and operating system versions -- see log
- [x] minimal example if possible: run pytest -v on pytest source

Full log: [pytest-7.1-test.txt](https://github.com/pytest-dev/pytest/files/8240636/pytest-7.1-test.txt)

```
[   32s] + pytest-3.9 --ignore=_build.python39 --ignore=_build.python310 --ignore=_build.python38 -v
[   34s] ============================= test session starts ==============================
[   34s] platform linux -- Python 3.9.10, pytest-7.1.0, pluggy-1.0.0 -- /usr/bin/python3.9
[   34s] cachedir: .pytest_cache
[   34s] hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/home/abuild/rpmbuild/BUILD/pytest-7.1.0/.hypothesis/examples')
[   34s] rootdir: /home/abuild/rpmbuild/BUILD/pytest-7.1.0, configfile: pyproject.toml, testpaths: testing
[   34s] plugins: forked-1.3.0, hypothesis-6.35.0, xdist-2.5.0
[   43s] collecting ... collected 3142 items
[   43s] 
...
[  516s] =================================== FAILURES ===================================
[  516s] ________________________ TestAssertionRewrite.test_name ________________________
[  516s] 
[  516s] self = <test_assertrewrite.TestAssertionRewrite object at 0x7f7610c6d040>
[  516s] request = <FixtureRequest for <Function test_name>>
[  516s] 
[  516s]     def test_name(self, request) -> None:
[  516s]         def f1() -> None:
[  516s]             assert False
[  516s]     
[  516s]         assert getmsg(f1) == "assert False"
[  516s]     
[  516s]         def f2() -> None:
[  516s]             f = False
[  516s]             assert f
[  516s]     
[  516s]         assert getmsg(f2) == "assert False"
[  516s]     
[  516s]         def f3() -> None:
[  516s]             assert a_global  # type: ignore[name-defined] # noqa
[  516s]     
[  516s]         assert getmsg(f3, {"a_global": False}) == "assert False"
[  516s]     
[  516s]         def f4() -> None:
[  516s]             assert sys == 42  # type: ignore[comparison-overlap]
[  516s]     
[  516s]         verbose = request.config.getoption("verbose")
[  516s]         msg = getmsg(f4, {"sys": sys})
[  516s]         if verbose > 0:
[  516s] >           assert msg == (
[  516s]                 "assert <module 'sys' (built-in)> == 42\n"
[  516s]                 "  +<module 'sys' (built-in)>\n"
[  516s]                 "  -42"
[  516s]             )
[  516s] E           AssertionError: assert 'assert sys == 42' == 'assert <modu...t-in)>\n  -42'
[  516s] E             + assert sys == 42
[  516s] E             - assert <module 'sys' (built-in)> == 42
[  516s] E             -   +<module 'sys' (built-in)>
[  516s] E             -   -42
[  516s] 
[  516s] testing/test_assertrewrite.py:210: AssertionError
[  516s] ___________ TestAssertionRewrite.test_dont_rewrite_if_hasattr_fails ____________
[  516s] 
[  516s] self = <test_assertrewrite.TestAssertionRewrite object at 0x7f7610c6da60>
[  516s] request = <FixtureRequest for <Function test_dont_rewrite_if_hasattr_fails>>
[  516s] 
[  516s]     def test_dont_rewrite_if_hasattr_fails(self, request) -> None:
[  516s]         class Y:
[  516s]             """A class whose getattr fails, but not with `AttributeError`."""
[  516s]     
[  516s]             def __getattr__(self, attribute_name):
[  516s]                 raise KeyError()
[  516s]     
[  516s]             def __repr__(self) -> str:
[  516s]                 return "Y"
[  516s]     
[  516s]             def __init__(self) -> None:
[  516s]                 self.foo = 3
[  516s]     
[  516s]         def f() -> None:
[  516s]             assert cls().foo == 2  # type: ignore[name-defined] # noqa: F821
[  516s]     
[  516s]         # XXX: looks like the "where" should also be there in verbose mode?!
[  516s]         msg = getmsg(f, {"cls": Y})
[  516s]         assert msg is not None
[  516s]         lines = msg.splitlines()
[  516s]         if request.config.getoption("verbose") > 0:
[  516s] >           assert lines == ["assert 3 == 2", "  +3", "  -2"]
[  516s] E           AssertionError: assert ['assert 3 ==...re Y = cls()'] == ['assert 3 ==...  +3', '  -2']
[  516s] E             At index 1 diff: ' +  where 3 = Y.foo' != '  +3'
[  516s] E             Full diff:
[  516s] E             - ['assert 3 == 2', '  +3', '  -2']
[  516s] E             + ['assert 3 == 2', ' +  where 3 = Y.foo', ' +    where Y = cls()']
[  516s] 
[  516s] testing/test_assertrewrite.py:283: AssertionError
[  516s] ________________________ TestAssertionRewrite.test_len _________________________
[  516s] 
[  516s] self = <test_assertrewrite.TestAssertionRewrite object at 0x7f76109e6e50>
[  516s] request = <FixtureRequest for <Function test_len>>
[  516s] 
[  516s]     def test_len(self, request) -> None:
[  516s]         def f():
[  516s]             values = list(range(10))
[  516s]             assert len(values) == 11
[  516s]     
[  516s]         msg = getmsg(f)
[  516s]         if request.config.getoption("verbose") > 0:
[  516s] >           assert msg == "assert 10 == 11\n  +10\n  -11"
[  516s] E           AssertionError: assert 'assert 10 ==..., 4, 5, ...])' == 'assert 10 ==...n  +10\n  -11'
[  516s] E               assert 10 == 11
[  516s] E             +  +  where 10 = len([0, 1, 2, 3, 4, 5, ...])
[  516s] E             -   +10
[  516s] E             -   -11
[  516s] 
[  516s] testing/test_assertrewrite.py:667: AssertionError
[  516s] ____________________ TestAssertionRewrite.test_custom_repr _____________________
[  516s] 
[  516s] self = <test_assertrewrite.TestAssertionRewrite object at 0x7f76109e6ac0>
[  516s] request = <FixtureRequest for <Function test_custom_repr>>
[  516s] 
[  516s]     def test_custom_repr(self, request) -> None:
[  516s]         def f() -> None:
[  516s]             class Foo:
[  516s]                 a = 1
[  516s]     
[  516s]                 def __repr__(self):
[  516s]                     return "\n{ \n~ \n}"
[  516s]     
[  516s]             f = Foo()
[  516s]             assert 0 == f.a
[  516s]     
[  516s]         msg = getmsg(f)
[  516s]         assert msg is not None
[  516s]         lines = util._format_lines([msg])
[  516s]         if request.config.getoption("verbose") > 0:
[  516s] >           assert lines == ["assert 0 == 1\n  +0\n  -1"]
[  516s] E           AssertionError: assert ['assert 0 ==... \\n~ \\n}.a'] == ['assert 0 == 1\n  +0\n  -1']
[  516s] E             At index 0 diff: 'assert 0 == 1\n +  where 1 = \\n{ \\n~ \\n}.a' != 'assert 0 == 1\n  +0\n  -1'
[  516s] E             Full diff:
[  516s] E             - ['assert 0 == 1\n  +0\n  -1']
[  516s] E             + ['assert 0 == 1\n +  where 1 = \\n{ \\n~ \\n}.a']
[  516s] 
[  516s] testing/test_assertrewrite.py:736: AssertionError
[  516s] =========================== short test summary info ============================
[  516s] FAILED testing/test_assertrewrite.py::TestAssertionRewrite::test_name - Asser...
[  516s] FAILED testing/test_assertrewrite.py::TestAssertionRewrite::test_dont_rewrite_if_hasattr_fails
[  516s] FAILED testing/test_assertrewrite.py::TestAssertionRewrite::test_len - Assert...
[  516s] FAILED testing/test_assertrewrite.py::TestAssertionRewrite::test_custom_repr
[  516s] ====== 4 failed, 3089 passed, 39 skipped, 10 xfailed in 483.64s (0:08:03) ======
```

**Comment by bluetech** [comment]
Thanks for the report.

I can reproduce this - the reason we didn't catch this failure is that it only happens when the pytest testsuite (of pytest itself) is run with `pytest -v`, which we usually don't. It's weird that we have tests that rely on the external pytest verbosity. It seems that something like this has already been fixed previously (2e7d6a6202bedf9555a02d1576adf789f49ae852) but regressed since it's not exercised.

These particular failures are due to fac8f284cd6ffd44c7d401093c67799766473be1 and likely don't indicate any issue, we just need to update the assertions to the new values. I will try to fix this tomorrow if no one beats me to it.

**Comment by nicoddemus** [comment]
Closed by #9768.
